addons/mail/static/src/new/thread/message.js

```diff
@@ -44,6 +44,13 @@ export class Message extends Component {
         return this.canBeDeleted;
     }
 
+    get showChannel() {
+        return (
+            this.message.resModel === "mail.channel" &&
+            !Number.isInteger(this.messaging.discuss.threadId)
```

Comment:
Don't rely on `threadId` being a number for it representing a channel.
Properly get corresponding thread and see whether it's a `mail.channel` in its data.

Reply:
Also this code is not conceptually correct:

This condition tells that whenever discuss app has a non-channel selected, then it shows "from message.recordName" on non-squashed message.

This is wrong: if a mailbox is selected in Discuss app and I leave discuss app, all chat window show the channel name on all messages of a channel:

<img width="342" alt="Screenshot 2022-11-23 at 00 11 23" src="https://user-images.githubusercontent.com/6569390/203440097-3008f6f5-0a6a-41fd-8cf9-71e3dfc3a696.png">

What you really want to do is to show the channel name of message when the origin thread is not the same as thread being shown on the message. You should pass `thread(Id)` as prop to this component, and compare them.

Doing so, the `on X` and `(from Y)` in the template are both conditional on showing a message in its non-origin thread. "on" vs. "from" is just a slight visual/textual difference of chatter vs. channel, but the underlying feature of explicitly showing the origin thread is exactly the same.